core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -866,6 +871,17 @@ public SqlNode toSql(@Nullable RexProgram program, RexNode rex) {
       }
     }
 
+    private SqlNode getSqlNodeByName(Context correlAliasContext, RexFieldAccess lastAccess) {
+      SqlNode node = null;
+      for (SqlNode sqlNode : correlAliasContext.fieldList()) {
```

Comment:
We can avoid unnecessary iteration of loop.

private SqlNode getSqlNodeByName(Context correlAliasContext, RexFieldAccess lastAccess) {
    for (SqlNode sqlNode : correlAliasContext.fieldList()) {
        if (isMatchingSqlNode(sqlNode, lastAccess)) {
            return sqlNode;
        }
    }
    return null;
}

Reply:
done